### PR TITLE
Add duration for schedule interval

### PR DESCRIPTION
**Pull request by jej**
This PR adds a duration for a schedule. It was one of my needs.

```
The function can be scheduled for a given period of time by adding
the keyword argument `_duration=` (in seconds). After this delay,
the function is unscheduled automagically.
```
Example:
```
import time
x = 0

def myfunc(dt):
    global x
    x += 1
    print(x)
    if x == 5:
        print(time.time() - start)

start = time.time()
pyglet.clock.schedule_interval(myfunc, 1, _duration=5)
```
Result:
```
1
2
3
4
5
5.000953674316406
```

It's seems reliable, but experimental and need to be tested before merging.

Hope it helps,
J.

### CODE REVIEW

**Comment by benmoran56** [comment]
This seems functionally fine, but I have some concerns.
The additionally checking inside of the clock each tick isn't ideal. This code needs to be as an fast as possible, and even tiny things like this can add up at 120Hz.
Another minor point is that the resolution/accuracy of the duration would be limited by the schedule frequency.

Instead of this, how about just using another schedule? 

```py
clock.schedule_interval(myfunc, 1.0)
clock.schedule_once(clock.unschedule, 5.0, myfunc)
```

**Comment by jej** [comment]
Oh sorry, I agree! Bad point to load more the scheduler at this stage... :|

Before that I unscheduled inside the scheduled function, by checking time at this level. Working well but not so DRY when many functions needs that kind of schedule. So I thought it was nice to have it in pyglet core without having to tweak the scheduled functions.

Trying your solution I get this:
```
Exception has occurred: TypeError
unschedule() takes 1 positional argument but 2 were given
  File ".../pyglet/clock.py", line 227, in call_scheduled_functions
    item.func(now - item.last_ts, *item.args, **item.kwargs)
  File ".../pyglet/app/base.py", line 212, in idle
    self.clock.call_scheduled_functions(dt)
  File ".../pyglet/app/base.py", line 151, in run
    timeout = self.idle()
              ^^^^^^^^^^^
  File ".../pyglet/app/__init__.py", line 76, in run
    event_loop.run(interval)
  File ".../application.py", line 57, in run
    pyglet.app.run()
  File ".../myapp.py", line 25, in <module>
    Application().run()
TypeError: unschedule() takes 1 positional argument but 2 were given
```

```
x = 0
def myfunc(dt):
    global x
    x += 1
    print(x)
    if x == 5:
        print(time.time() - start)

start = time.time()
pyglet.clock.schedule_interval(myfunc, 1)
pyglet.clock.schedule_once(pyglet.clock.unschedule, 5.0, myfunc)
```

**Comment by jej** [comment]
Ok, I manage, don't worry...

**Comment by jej** [comment]
Ok, I managed by the way you mention, quite better of course! Needed some tweak to accept the dt argument.

Now, what about the `_duration` args to achieve this? It's not possible the have a default nulled arguments, because of the kwargs. Another solution could be a new method like `schedule_interval_for_duration(myfunc, 1, 5)`. But it we cannot create a new method for each functionality...

**Comment by jej** [comment]
And feel free to decline the request if the contribution is not useful for pyglet.

It's just a pleasure for me to make things with and look into it!

**Comment by benmoran56** [comment]
I think unfortunately the extra checks involved make this a no-go for the moment. I'm also not really sure how people will feel about the kwargs hack either.

**Comment by jej** [comment]
There is no more extra check after my last commit. Did you notice the change?

I agree the _duration is not the best way, because not a practice in pyglet.

But the feature itself is quite useful. Launching a regular task for a given period of time, without having to manage it manually...

**Comment by jej** [comment]
Yep, according to the current pyglet design pattern, a new method would do the job.

Ok then, I will commit something later.

**Comment by jej** [comment]
PS: the function is called `schedule_interval_/for/_duration` to avoid confusion with `interval_duration`

File: pyglet/clock.py [hunk 1]
```diff
@@ -458,6 +458,28 @@ def schedule_interval(self, func, interval, *args, **kwargs):
...
+        (see `schedule_interval`) and unschedule it after `duration` seconds.
+
+        The callback function prototype is the same as for `schedule`.
+
+        :Parameters:
+            `func` : callable
+                The function to call when the timer lapses.
+            `interval` : float
+                The number of seconds to wait between each call.
+            `duration=` : float
```
Comment (benmoran56) [inline pyglet/clock.py:472]: Just a tiny typo here (=).

**Review by benmoran56 — CHANGES_REQUESTED** [review]
Looks good to me. Just a tiny typo, but after that it should be good to merge.

File: pyglet/clock.py (hunk 1 above)
Comment (jej) [reply pyglet/clock.py]: ooops sorry, a remain of the keyword argument documentation...

thx

**Review by benmoran56 — APPROVED** [review]

**Pull request merged**